src/dataflow/operators/map_operator.rs

```diff
@@ -1,99 +1,78 @@
-use crate::dataflow::message::Message;
 use crate::dataflow::{
-    stream::WriteStreamT, Data, Operator, OperatorConfig, ReadStream, Timestamp, WriteStream,
+    context::OneInOneOutContext, message::Message, operator::OneInOneOut, stream::WriteStreamT,
+    Data,
 };
 use serde::Deserialize;
-use std::marker::PhantomData;
+use std::sync::Arc;
+
+pub trait MapFn<D1, D2>: Fn(&D1) -> D2 + Send + Sync {}
+
+impl<D1, D2, F: Fn(&D1) -> D2 + Send + Sync> MapFn<D1, D2> for F {}
 
 /// An operator that maps an incoming stream of type D1 to a stream of type D2 using the provided
 /// function.
 ///
 /// # Example
-/// The below example shows how to use a MapOperator to double an incoming stream of u32 messages,
-/// and return them as u64 messages.
+/// The below example shows how to use a MapOperator to square an incoming stream of usize messages,
+/// and return them.
 ///
 /// ```
-/// # use erdos::dataflow::{stream::IngestStream, operators::MapOperator, OperatorConfig};
-/// # use erdos::*;
-/// #
-/// # let mut u32_stream = IngestStream::new(0);
-/// #
-/// // Add the mapping function as an argument to the operator via the OperatorConfig.
-/// let map_config = OperatorConfig::new()
-///     .name("MapOperator")
-///     .arg(|data: &u32| -> u64 { (data * 2) as u64 });
-/// let u64_stream = connect_1_write!(MapOperator<u32, u64>, map_config, u32_stream);
+/// fn square_fn(a: &usize) -> usize {
+///     return a * a;
+/// }
+/// # Add the mapping function as an argument to the operator via the OperatorConfig.
+/// let map_config = OperatorConfig::new().name("MapOperator");
+/// let map_stream = erdos::connect_one_in_one_out(
+///     || -> MapOperator<usize, usize> { MapOperator::new(square_fn) },
```

Comment:
I'm not exactly sure if Rust's semantics allow the squaring of a `usize` value to return a value of type `usize`?